### DISCUSSION TITLE
ESP32 power saving options, modem sleep/slowing cpu.

### DISCUSSION THREAD

**Discussion by ThinkTransit** [micropython/micropython#10861] [ESP32]
This was raised on discord a while ago by somebody else, bringing it here for more exposure.

How do you implement modem sleep on the esp32 chips and is it possible to also slow down the cpu without breaking micropython to achieve the following ~2-4ma power consumption?

![image](https://user-images.githubusercontent.com/74940387/221390849-007452a3-208c-4e8d-a34e-c236880b0f10.png)

If nothing has been done in this space I might try and give it a go.

Any thoughts?

**Comment by davefes** [comment]
https://micropython-tve.readthedocs.io/en/esp32-low-power/library/esp32-lowpower.html

says you can drop the CPU frequency to 20MHz.  Maybe listen_interval is interesting.

**Reply by ThinkTransit** [reply]
Thanks that's very interesting, looks like you can potentially go down to 10MHz by using the min freq option.

**Comment by karfas** [comment]
There is a patch request - #8993 - to enable the WIFI power save option on ESP32.
